app/scripts/components/common/mapbox/map-options/use-basemap.ts

```diff
@@ -0,0 +1,100 @@
+import { Layer, Style } from 'mapbox-gl';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import {
+  BasemapId,
+  BASEMAP_STYLES,
+  getStyleUrl,
+  GROUPS_BY_OPTION,
+  Option
+} from './basemaps';
+
+export function useBasemap() {
+  const [basemapStyleId, setBasemapStyleId] = useState<BasemapId>('satellite');
+
+  const onBasemapStyleIdChange = useCallback((basemapId) => {
+    setBasemapStyleId(basemapId);
+  }, []);
+
+  const [baseStyle, setBaseStyle] = useState<Style | undefined>(undefined);
+
+  useEffect(() => {
+    const mapboxId = basemapStyleId
+      ? BASEMAP_STYLES.find((b) => b.id === basemapStyleId)!.mapboxId
+      : BASEMAP_STYLES[0].mapboxId;
+
+    const url = getStyleUrl(mapboxId);
+    const controller = new AbortController();
+
+    const load = async () => {
+      const styleRaw = await fetch(url, { signal: controller.signal });
+      const styleJson = await styleRaw.json();
+      setBaseStyle(styleJson as Style);
+    };
+    load();
+    return () => {
+      controller.abort();
+    };
+  }, [basemapStyleId]);
+
+  const [labelsOption, setLabelsOption] = useState(true);
+  const [boundariesOption, setBoundariesOption] = useState(true);
+  const onOptionChange = useCallback(
+    (option: Option, value: boolean) => {
+      if (option === 'labels') {
+        setLabelsOption(value);
+      } else {
+        setBoundariesOption(value);
+      }
+    },
+    [setLabelsOption, setBoundariesOption]
+  );
+
+  // Apply labels and boundaries options, by setting visibility on related layers
+  // For simplicity's sake, the Mapbox layer group (as set in Mapbox Studio) is used
+  // to determine wehether a layer is a labels layer or boundaries or none of those.
```

Comment:
Should we document (in some dev doc), what is needed if we want to add a new base style? It seems that certain groups are required?! Or am I not getting this the right way?

Reply:
Yes, you are right. I think the right place to document is directly in that file or basemap.ts.
Another option is to migrate the Mapbox styles directly in the source code, and store those groups as constants that we can reuse directly within the styles. But then we loose the ability to edit them within Mapbox Studio.